assets/javascripts/discourse/components/show-original-content.gjs

```diff
@@ -0,0 +1,39 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+
+export default class ShowOriginalContent extends Component {
+  @service router;
+  @tracked active = true;
+
+  constructor() {
+    super(...arguments);
+    this.active = !new URLSearchParams(window.location.search).has("show");
+  }
+
+  @action
+  async showOriginal() {
+    this.active = !this.active;
+    window.location.search = this.active ? "" : `show=original`;
```

Comment:
Is it possible that there will be params already? Maybe it would be safer to append params?

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append